blog-posts/getting-started-gatsby-next-remix.md

```diff
@@ -0,0 +1,306 @@
+
+```tsx
+const BlogPost = ({data}) => {
+    return (
+        <div dangerouslySetInnerHTML={{__html: data.markdownRemark.html}}/>
+    )
+}
+
+export const query = graphql`
+    query ($id: String) {
+        markdownRemark(id: {eq: $id}) {
+            frontmatter {
+                title
+                path
+            }
+            html
+        }
+    }
+`
+
+export default BlogPost
+```
+
+
+> 💡 Alternatively, pages can also be created in `gatsby-node.js`. For this, the `createPages` API provided by Gatsby must be used. For more information on this, as well as an easy-to-understand example, see the [Gatsby documentation](https://www.gatsbyjs.com/docs/creating-and-modifying-pages/#creating-pages-in-gatsby-nodejs).
+
+
+#### **Next.js**
+
+To create a static generated page in Next.js the `async` function `getStaticPaths` has to be exported. `getStaticPaths` must return an object with the keys:
+
+- `path`: This key determines which path will be pre-rendered. In the in-code comment below the structure of `path` is given as an example.
+- `fallback`: You can set the value of fallback to `true`, `false` or `blocking`
+    - `false` → Any paths not returned by `getStaticPaths` will end up in a 404 page
+    - `true` → On the first request to a not generated path, Next.js will serve a fallback version of the page, instead of a 404 page. The HTML and JSON of this path is now generated in the background and and then the JSON is sent to the requesting browser. Now the fallback page changes into the full page. Next.js will add this path to the pre-rendered pages in order to be able to provide the already generated page for further requests.
+      This is useful if you have a page with a lot of static pages because then not all pages have to be generated at built time, which makes the built much faster.
+    - `blocking` → Same procedure as for `fallback: true`, except that there is no fallback while the HTML is being generated. There are only very special uses cases in which this makes sense, such as AMP.
+
+Remember to name the file after the relevant key from the `params` object with square brackets, in our case `[path].tsx`.
+To load the data for a blogpost the function `getStaticProps` is needed. This function now receives `{params}`, which can be used to determine the path and the depending data. Finally, in the actual page component, the data can be accessed via props (here via `{postData}`).
+
+```tsx
+export async function getStaticPaths() {
+    //     paths: [
+    //         {params: {path: 'first-blogpost'}},
+    //         {params: {path: 'second-blogpost'}},
+    //         ...
+    //         {params: {path: 'fifth-blogpost'}}
+    //     ],
+    const paths = getAllPostPaths()
```

Comment:
I'm confused. Should I look at the comment or ignore it? What is getAllPostPaths about ?